CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
+
+project(CppExtraSamples)
+
+set(CMAKE_CXX_STANDARD 11)
+set(ZIVID_VERSION 1.5.0)
+
+option(USE_EIGEN3 "Enable samples which depend on Eigen 3" ON)
+option(USE_OPENCV "Enable samples which depend on OpenCV" ON)
+option(USE_PCL "Enable samples which depend on Point Cloud Library (PCL)" ON)
+option(USE_VIS3D "Enable samples which depend on the Zivid Cloud Visualizer. These can be hard to get to build on Linux due to a specific Qt dependency" ON)
+
+find_package(Zivid ${ZIVID_VERSION} COMPONENTS Core REQUIRED)
+find_package(PCL 1.2 QUIET)
+find_package(OpenCV 4.0.1 QUIET OPTIONAL_COMPONENTS core highgui)
+
+set(SAMPLES
+    Zivid/CaptureHDRCompleteSettings
+    Zivid/CaptureHDRLoop
+    Zivid/CaptureSavePLY
+    Zivid/ConnectToSerialNumberCamera
+    Zivid/GetCameraIntrinsics
+    Zivid/ReadZDF
+    Zivid/ZDF2PLY
+    ZividPCL/ReadPCLVis3D
+    ZividPCL/CaptureWritePCLVis3D
+    ZividPCL/CaptureFromFileWritePCLVis3D
+    ZividPCL/ZDF2PCD
+    ZividOpenCV/ZDF2OpenCV
+    Downsample)
+
+set(Eigen3_DEPENDING Downsample)
```

Comment:
There is no (guaranteed) correlation between this lists and the directory convention used above? The list below is more flexible than what can be expressed in the folder structure. Where are new samples supposed to be placed? For example one that depends on OpenCV and PCL.

Reply:
I discussed that with @SatjaSivcev a few days ago and agreed it's outside the scope of this PR. With this PR in place, they are now free to change the folder structure as they like.